sagemaker_neo_compilation_jobs/mxnet_mnist/mxnet_mnist_neo.ipynb

```diff
@@ -6,11 +6,22 @@
    "source": [
     "## MNIST Training, Compilation and Deployment with MXNet Module and Sagemaker Neo\n",
     "\n",
-    "The **SageMaker Python SDK** makes it easy to train and deploy MXNet models. In this example, we train a simple neural network using the Apache MXNet [Module API](https://mxnet.apache.org/api/python/module/module.html) and the MNIST dataset. The MNIST dataset is widely used for handwritten digit classification, and consists of 70,000 labeled 28x28 pixel grayscale images of hand-written digits. The dataset is split into 60,000 training images and 10,000 test images. There are 10 classes (one for each of the 10 digits). The task at hand is to train a model using the 60,000 training images and subsequently test its classification accuracy on the 10,000 test images. \n",
+    "The **SageMaker Python SDK** makes it easy to train, compile and deploy MXNet models. In this example, we train a simple neural network using the Apache MXNet [Module API](https://mxnet.apache.org/versions/1.5.0/api/python/module/module.html) and the MNIST dataset. The MNIST dataset is widely used for handwritten digit classification, and consists of 70,000 labeled 28x28 pixel grayscale images of hand-written digits. The dataset is split into 60,000 training images and 10,000 test images. There are 10 classes (one for each of the 10 digits). The task at hand is to train a model using the 60,000 training images, compile the trained model using SageMaker Neo and subsequently test its classification accuracy on the 10,000 test images. \n",
     "\n",
     "### Setup\n",
     "\n",
-    "First we need to define a few variables that will be needed later in the example."
+    "First we need to upgrade the [SageMaker SDK for Python](https://sagemaker.readthedocs.io/en/stable/v2.html) if it is not and define a few variables that will be needed later in the example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sagemaker\n",
```

Comment:
Some folks are still using Python SDK v<2. Since we are installing v>=2, make them revert back to their original version at the end of the notebook so as to not mess up their environment. Something like: 

In the beginning of the notebook: 
```
import sagemaker

# Get original version:
original_version = sagemaker.__version__
# update sagemaker
if original_version.split('.')[0] == '1':
    !~/anaconda3/envs/mxnet_p36/bin/pip install --upgrade sagemaker
.
.
.
```
At the end of the notebook, add:
```
.
.
.
# Revert back to their original Python SDK
s = 'sagemaker=={}'.format(original_version)
print("Rolling back to... {}".format(s))
%pip install {s}
import sagemaker
print("{} installed!".format(sagemaker.__version__))
```

Reply:
Importing `sagemaker` before `pip install --upgrade` has the following issue:
The customer has to restart the kernel for the new version to take effect. This means that a kernel restart is required before executing all the cells in the notebook.
How do we address this @ngluna ?

Reply:
Addressed the changes after discussing offline